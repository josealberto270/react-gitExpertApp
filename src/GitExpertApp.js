import React, { useEffect, useState } from 'react'
import { AddCategoria } from './components/AddCategoria'
import { GifGrid } from './components/GifGrid'

export const GifExpertApp = ({ defaultCategorias = [] }) => {

  // const categorias = ['One Punch', 'Samurai X', 'Dragon Ball']
  const [categorias, setCategorias] = useState(defaultCategorias)

  // const handleAdd = () => {
  //   const newCategoria = 'One Piece'
  //   setCategorias([...categorias,newCategoria])
  // }

  return <>
    <h2>GifExpertApp</h2>
    <AddCategoria setCategorias={setCategorias} />
    <hr />
    <ol>
      {
        categorias.map((categoria, index) =>
          <GifGrid key={index} categoria={categoria} />
        )
      }
    </ol>
  </>
}