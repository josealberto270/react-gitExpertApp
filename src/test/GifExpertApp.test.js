import { shallow } from 'enzyme';
import React from 'react';
import '@testing-library/jest-dom'
import { GifExpertApp } from '../GitExpertApp';


describe('Prueba de la pagina GifExpertApp', () => {

  test('debe de mostrar <GifExpertApp />', () => {

    const wrapper = shallow(<GifExpertApp />)
    expect(wrapper).toMatchSnapshot()

  })

  test('debe de mostrar una lista de categotras', () => {
    const categorias = ['Naruto', 'Dragon ball']
    const wrapper = shallow(<GifExpertApp defaultCategorias={categorias} />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('GifGrid').length).toBe(categorias.length)

  })


})
