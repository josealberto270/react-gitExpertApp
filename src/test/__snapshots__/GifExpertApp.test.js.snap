// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de la pagina GifExpertApp debe de mostrar <GifExpertApp /> 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategoria
    setCategorias={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Prueba de la pagina GifExpertApp debe de mostrar una lista de categotras 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategoria
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categoria="Naruto"
      key="0"
    />
    <GifGrid
      categoria="Dragon ball"
      key="1"
    />
  </ol>
</Fragment>
`;
